src/main/java/wl/exception/ValidationException.java

```diff
@@ -0,0 +1,18 @@
+package wl.exception;
+
+import java.util.List;
+
+/**
+ * Validation exception.
+ */
+public class ValidationException extends RuntimeException {
+
+  /**
+   * Constructor.
+   *
+   * @param violations List of errors.
+   */
+  public ValidationException(final List<String> violations) {
+    super("Validation errors: " + String.join("; ", violations));
+  }
+}
```

Comment:
If you want to access errors later, you may store them:
```suggestion
public class ValidationException extends RuntimeException {
  private final List<String> violations;

  /**
   * Constructor.
   *
   * @param violations List of errors.
   */
  public ValidationException(final List<String> violations) {
    super("Validation errors: " + String.join("; ", violations));
    this.violations = violations;
  }
  
  public List<String> getViolations() {
    return violations;
  }
}
```

Also, you can use Map:
```java
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class UserRegistrationValidationException extends RuntimeException {
    private final Map<String, List<String>> errors;

    public UserRegistrationValidationException(Map<String, List<String>> errors) {
        super("User registration validation failed");
        this.errors = new HashMap<>(errors);
    }

    public Map<String, List<String>> getErrors() {
        return errors;
    }

    @Override
    public String getMessage() {
        return super.getMessage() + ": " + errors.toString();
    }
}
```